### ISSUE TITLE
Obsolete diagnostic not reported for type named `var` brought by `using static`

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#64713]
Originally reported on [Discord](https://discord.com/channels/143867839282020352/598678594750775301/1029980662108782632)

```csharp
using static Ext;
using System;

var a = new(); // This should produce an error.

class Ext {
    [Obsolete("error", true)]
    public class var {
    }
}
```

**Comment by 333fred** [comment]
Note that this _only_ occurs when the type is named `var`. Even `dynamic` is ok.

**Comment by Youssef1313** [comment]
It also has nothing to do with `using static` specifically.

```
using System;

var a = new();

[Obsolete("error", true)]
public class var
{
}
```

The issue reproduces with above code.
